libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -215,16 +215,11 @@ bool AP_Terrain::height_above_terrain(float &terrain_altitude, bool extrapolate)
     if (!height_terrain_difference_home(terrain_difference, extrapolate)) {
         return false;
     }
-    Location loc;
-    if (!ahrs.get_position(loc)) {
```

Comment:
I think we should leave this check that we know our current location.  It looks to me like the get_relative_position_D_home returns the alt-above-ekf-origin if it can't find the home location.  Those two things are really not equivalent so I don't think we should be doing that.  They are normally quite close but they can be off by a few meters or even hundreds of meters if the user moves the home location.